tests/flows/test_vesting.py

```diff
@@ -106,73 +103,535 @@ def test_main(
+    assert bpool.get_swap_fee() == to_wei("0.003")
+    assert bpool.community_fee(get_address_of_type(config, "Ocean")) == 0
+    assert bpool.community_fee(erc20_token.address) == 0
+    assert bpool.publish_market_fee(get_address_of_type(config, "Ocean")) == 0
+    assert bpool.publish_market_fee(erc20_token.address) == 0
+    assert (
+        erc20_token.balanceOf(get_address_of_type(config, "Staking"))
+        == MAX_UINT256 - initial_ocean_liq
+    )
+
+    # check if the vesting amount is correct
+    assert side_staking.get_vesting_amount(erc20_token.address) == vesting_amount
+
+    # check if vesting is correct
+    dt_balance_before = erc20_token.balanceOf(erc20_token.get_payment_collector())
+    available_vesting_before = side_staking.get_available_vesting(erc20_token.address)
+
+    # advance 5 blocks to see if available vesting increased
+    for _ in range(4):
+        # send dummy transactions to increase block count (not part of the actual functionality)
+        send_ether(consumer_wallet, ZERO_ADDRESS, 1)
+
+    available_vesting_after = side_staking.get_available_vesting(erc20_token.address)
+    assert (
+        available_vesting_after > available_vesting_before
+    ), "Available vesting was not increased!"
+    tx_hash = side_staking.get_vesting(erc20_token.address, consumer_wallet)
+    vested_amount = side_staking.get_amount_vested_from_event(
+        tx_hash=tx_hash, erc20_token=erc20_token, from_wallet=consumer_wallet
+    )
+
+    # 5 blocks passed, vesting started block = block_deployed_pool (deploy pool tx)
+    target_vesting_amount = (5 * vesting_amount) // (
+        side_staking.get_vesting_end_block(erc20_token.address) - block_deployed_pool
+    )
+    assert vested_amount == target_vesting_amount
+    assert vested_amount == side_staking.get_vesting_amount_so_far(erc20_token.address)
+    assert dt_balance_before < erc20_token.balanceOf(pmt_collector)
+    assert erc20_token.balanceOf(pmt_collector) == vested_amount
+
+
+@pytest.mark.unit
+def test_vesting_progress(
+    web3,
+    config,
+    publisher_wallet,
+    factory_router,
+    side_staking,
+):
+    """Tests vesting progress after some percentages of the vested passed blocks."""
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, publisher_wallet, publisher_wallet, cap=to_wei(1000)
     )
+    mint_fake_OCEAN(config)
+    initial_ocean_liq = to_wei(200)
+    vested_blocks = factory_router.get_min_vesting_period()
+    percentages = [0.00000005, 0.000003, 0.00001]
+    checkpoint_blocks = list(map(lambda p: int((vested_blocks * p) / 100), percentages))
+
+    vesting_amount = initial_ocean_liq // 100 * 9
+
+    block_before_deploying_pool = web3.eth.block_number
+    bpool = _deploy_ocean_pool(
+        erc20_token=erc20_token,
+        factory_router=factory_router,
+        config=config,
+        web3=web3,
+        from_wallet=publisher_wallet,
+    )
+    assert bpool.is_finalized()
```

Comment:
Remove line, out of scope for this function's testing

Reply:
Removed all the occurrences